src/app/Shared/Services/Api.service.tsx

```diff
@@ -90,15 +90,18 @@ export class ApiService {
     );
   }
 
-  createRecording(
-    { recordingName, events, duration  }: { recordingName: string; events: string; duration?: number }
+  createRecording({ recordingAttributes }: { recordingAttributes: RecordingAttributes }
```

Comment:
This is the place where the function definition should be simplified. This function should accept simply a `RecordingAttributes` parameter, not an object with a single `recordingAttributes` key of type `RecordingAttributes`.